Sources/ComposableArchitecture/Instrumentation.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+
+/// Interface to enable tracking/instrumenting the activity within TCA as ``Actions`` are sent into ``Store``s and
+/// ``ViewStores``, ``Reducers`` are executed, and ``Effects`` are observed.
+///
+/// The way the library will call the closures provided is identical to the way that the ``Actions`` and ``Effects`` are
+/// handled internally. That means that there is likely to be ``Instrumentation.ViewStore`` `will|did` pairs contained
+/// within the bounds of an ``Instrumentation.Store`` `will|did` pair. For example: Consider sending a simple ``Action``
+/// into a ``ViewStore`` that does not produce any synchronous ``Effects`` to be generated by a ``Reducer``, and the
+/// ``ViewStore`` is scoped off a parent ``Store``s state:
+/// ```
+/// ViewStore.send(.someAction)
+/// .pre, .viewStoreSend
+///   Store.send(.someAction)
+///   .pre, .storeSend
+///     The Store will begin processing .someAction
+///     .pre, .storeProcessEvent
+///       The Store's reducer handles .someAction
+///       Any returned actions from the reducer are queued up
+///     .post, .storeProcessEvent
+///     The above(willProcess -> didProcess) is repeated for each queued up action within Store
+///     .pre, .storeChangeState
+///       The Store updates its state
+///       Any Stores based off the parent Store have their states updated, there for the below may be called multiple times
+///       .pre, .viewStoreDeduplicate for impacted ViewStores
+///       .post, .viewStoreDeduplicate
```

Comment:
can/does anything happen between .pre and .post for .viewStoreDeduplicate, or would this be better reported as a single duration value post-facto? IDK, and probably not worth changing now, but maybe worth thinking about during the upstreaming convo.

Reply:
Yes, the equality comparison happens. And that just happens to be something we have not upstreamed either